### PR TITLE
Fetch data for elastic farm v2

### PR DESCRIPTION
**Pull request by viet-nv**

### CODE REVIEW

**Review by nguyenhoaidanh — APPROVED** [review]

File: src/state/farms/elasticv2/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+`
+
+let isSubscribed = false
+
+const positionManagerInterface = new Interface(NFTPositionManagerABI.abi)
+
+export const useElasticFarmsV2 = (subscribe = false) => {
+  const dispatch = useAppDispatch()
+  const { networkInfo, isEVM, chainId, account } = useActiveWeb3React()
+  const elasticFarm = useAppSelector(state => state.elasticFarmV2[chainId || 1] || defaultChainData)
```
Comment (namgold) [inline src/state/farms/elasticv2/hooks.ts:80]: `|| 1` :pepe-sad:
```suggestion
  const elasticFarm = useAppSelector(state => state.elasticFarmV2[chainId || ChainId.MAINNET] || defaultChainData)
```

Comment (viet-nv) [reply src/state/farms/elasticv2/hooks.ts:80]: changed

File: src/state/farms/elasticv2/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        isRemoved
+        tickUpper
+        tickLower
+        weight
+      }
+    }
+  }
+`
+
+let isSubscribed = false
```
Comment (namgold) [inline src/state/farms/elasticv2/hooks.ts:73]: move inside component and make it a useRef?

File: src/state/farms/elasticv2/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+  useEffect(() => {
+    if (error && chainId) {
+      dispatch(setFarms({ chainId, farms: [] }))
+      dispatch(setLoading({ chainId, loading: false }))
+    }
+  }, [error, dispatch, chainId])
+
+  useEffect(() => {
+    if (data?.farmV2S && chainId) {
```
Comment (namgold) [inline src/state/farms/elasticv2/hooks.ts:125]: no need to check chainId
```suggestion
    if (data?.farmV2S) {
```

File: src/state/farms/elasticv2/hooks.ts (hunk 2 above)
Comment (viet-nv) [reply src/state/farms/elasticv2/hooks.ts:73]: no, I want this is global, not only access by one component.
for example, when 2 components use this hook, it's only setInterval 1 time

**Review by XiaoYhun — APPROVED** [review]

**Pull request merged**